com.unity.shadergraph/Editor/Drawing/Blackboard/BlackboardProvider.cs

```diff
@@ -247,17 +248,39 @@ void AddBuiltinKeyword(GenericMenu gm, ShaderKeyword keyword)
         void EditTextRequested(Blackboard blackboard, VisualElement visualElement, string newText)
         {
             var field = (BlackboardField)visualElement;
+            var fieldView = (BlackboardFieldView)visualElement;
             var input = (ShaderInput)field.userData;
             if (!string.IsNullOrEmpty(newText) && newText != input.displayName)
             {
                 m_Graph.owner.RegisterCompleteObjectUndo("Edit Graph Input Name");
                 input.displayName = newText;
                 m_Graph.SanitizeGraphInputName(input);
                 field.text = input.displayName;
+                // need to trigger the inspector update to match
+                fieldView.InspectorUpdateTrigger();
                 DirtyNodes();
             }
         }
 
+        void UpdateBlackboardView()
+        {
+            var selection = new List<ISelectable>();
```

Comment:
Can you can pass `blackboard.selection` directly to the constructor? It might throw if the collection is null but if not it's a little cleaner.

Reply:
can do, this never gets called in a scenario when the blackboard selection is null anyway /shrug